src/Module/Update/Channel.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * See update_profile.php for documentation
+ */
+
+namespace Friendica\Module\Update;
+
+use Friendica\Content\Conversation;
+use Friendica\Core\System;
+use Friendica\DI;
+use Friendica\Module\Conversation\Channel as ChannelModule;
+
+/**
+ * Asynchronous update module for the Channel page
+ *
+ * @package Friendica\Module\Update
+ */
+class Channel extends ChannelModule
+{
+	protected function rawContent(array $request = [])
+	{
+		$this->parseRequest();
+
+		$o = '';
+		if (!empty($request['force'])) {
+			$o = DI::conversation()->render(self::getItems(), Conversation::MODE_CHANNEL, true, false, 'created', DI::userSession()->getLocalUserId());
```

Comment:
Please remove the `DI` dependency by declaring an explicit constructor and adding parameters with the type returned by the various `DI` calls in this class. Then you can save the parameters to local variables and use these variables instead of the static calls.

Due to inheritance with the Channel module, these variables might already be available here.

Reply:
Changed